src/conditions/check_fields.rs

```diff
@@ -523,6 +524,12 @@ inventory::submit! {
     ConditionDescription::new::<CheckFieldsConfig>("check_fields")
 }
 
+impl GenerateConfig for CheckFieldsConfig {
+    fn generate_config() -> toml::Value {
+        toml::Value::try_from(&Self::default()).unwrap()
+    }
+}
+
```

Comment:
Could much of this copy-and-paste be avoided with a blanket implementation?

```rust
impl<T: Default> GenerateConfig for T {
    fn generate_config() -> toml::Value {
        toml::Value::try_from(&Self::default()).unwrap()
    }
}
```

Reply:
This would not allow implementing custom GenerateConfig on type that already has `Default`.

I added a macro (impl_generate_config_from_default), and a blanket implementation of GenerateConfig; 
```rust

pub trait GenerateConfig {
    fn generate_config() -> toml::Value {
        toml::Value::Table(Default::default())
    }
}
```

These two reduce copy-paste a lot.

Reply:
Ah, I thought I had proved this out in the playground, but I was missing a trait that broke it like you indicated. This would have required the specialization feature which has not yet stabilized.